next.config.mjs

```diff
@@ -24,6 +24,9 @@ const nextJSConfig = () => {
     distDir: 'client/www/next-build',
     env: {
       BUILD_ENV: process.env.BUILD_ENV,
+      ALGOLIA_APP_ID: process.env.ALGOLIA_APP_ID,
+      ALGOLIA_API_KEY: process.env.ALGOLIA_API_KEY,
+      ALGOLIA_INDEX_NAME: process.env.ALGOLIA_INDEX_NAME,
```

Comment:
In the `LayoutHeader.tsx`, we try to use these the env variables for Algolia if they exist and if they don't we default to the ones here https://github.com/aws-amplify/docs/blob/main/src/constants/algolia.ts, which returns our production results.

To facilitate testing on staging sites we need to pass them into the next.config so that it's passed to our nextjs app

Reply:
How does the fallback work?

Reply:
nevermind, see it here: https://github.com/aws-amplify/docs/blob/4c8a71162c7b7a0df15627c951e43fd570178a39/src/components/Layout/LayoutHeader.tsx#L79

Reply:
Yea I think in order to test out our dev index, we'll have to set it up so that it can take in the env vars that are set in hosting.

Reply:
so, `process.env.ALGOLIA_INDEX_NAME` will come from Amplify Hosting if it's available, right?

Reply:
yea that's correct